### ISSUE TITLE
Sample does not rock

### ISSUE THREAD

**Issue by SimpleSamples** [octokit/octokit.graphql.net#193]
I installed into a console project using NuGet. I created an OAuth token as in [Forming Calls with GraphQL](https://developer.github.com/v4/guides/forming-calls/) and [Creating a personal access token for the command line](https://help.github.com/en/articles/creating-a-personal-access-token-for-the-command-line). The following is my code.

```cs
using Octokit.GraphQL;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using static Octokit.GraphQL.Variable;

namespace ConsoleGitHubGraphQL
{
    class Program
    {
        static void Main(string[] args)
        {
            AsyncMethodAsync();
        }

        static async void AsyncMethodAsync()
        {
            // The product name is either the product, the GitHub
            // Organization, or the GitHub username.
            var productInformation = new ProductHeaderValue("SimpleSamples" /*, "YOUR_PRODUCT_VERSION"*/);
            var connection = new Connection(productInformation, "myOAuthtoken");

            var query = new Query()
                .RepositoryOwner(Var("owner"))
                .Repository(Var("name"))
                .Select(repo => new
                {
                    repo.Id,
                    repo.Name,
                    repo.Owner.Login,
                    repo.IsFork,
                    repo.IsPrivate,
                }).Compile();

            var vars = new Dictionary<string, object>
                {
                    { "owner", "SimpleSamples" },
                    { "name", "SimpleSamples.Repos" },
                };

            try
            {
                var result = await connection.Run(query, vars);
                Console.WriteLine(result.Login + " & " + result.Name + " Rocks!");
            }
            catch (Exception ex)
            {
                Console.WriteLine("Error: {0}", ex.Message);
            }

        }
    }
}
```

It works up to the call to connection.Run, then quits. It just ends. When I single-step using the debugger it still ends at that line, no diagnostic information or anything.

I apologize if this is a C# problem or a problem resulting from me not reading the documentation well enough. Is it? Or should it **at least** give me an exception?

**Comment by jcansdale** [comment]
It looks like the problem is here:
```cs
        static void Main(string[] args)
        {
            AsyncMethodAsync();
        }
```

You need to wait for the `AsyncMethodAsync` method to complete. Try the following instead:

```cs
using Octokit.GraphQL;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using static Octokit.GraphQL.Variable;

namespace ConsoleGitHubGraphQL
{
    class Program
    {
        static async Task Main(string[] args)
        {
            await AsyncMethodAsync();
        }

        static async Task AsyncMethodAsync()
        {
            // The product name is either the product, the GitHub
            // Organization, or the GitHub username.
            var productInformation = new ProductHeaderValue("SimpleSamples" /*, "YOUR_PRODUCT_VERSION"*/);
            var connection = new Connection(productInformation, "YOUR PERSONAL ACCESS TOKEN");

            var query = new Query()
                .RepositoryOwner(Var("owner"))
                .Repository(Var("name"))
                .Select(repo => new
                {
                    repo.Id,
                    repo.Name,
                    repo.Owner.Login,
                    repo.IsFork,
                    repo.IsPrivate,
                }).Compile();

            var vars = new Dictionary<string, object>
                {
                    { "owner", "github" },
                    { "name", "VisualStudio" },
                };

            var result = await connection.Run(query, vars);
            Console.WriteLine(result.Login + " & " + result.Name + " Rocks!");
        }
    }
}
```

**Comment by jcansdale** [comment]
Alternatively you could do:

```
        static void Main(string[] args)
        {
            AsyncMethodAsync().Wait();
        }

        static async Task AsyncMethodAsync()
        {
```

If you're compiling for an older version of C#.

**Comment by jcansdale** [comment]
I'm closing this because I belied a missing `await AsyncMethodAsync()` was the issue.

**Comment by SimpleSamples** [comment]
I apologize for not saying that the sample code is from [Bringing GraphQL to Octokit.NET](https://github.blog/2018-07-13-graphql-for-octokit) except it has problems that I attempted to fix. Is that article within the scope of here? Note that the message at the bottom of that sample says "Rocks" and that is why I made the title of this "Sample does not rock".

That sample has a:

`using Octokit.GraphQL.Core;`

That does not work or at least is not necessary. The sample also says Connection.Run instead of connection.Run. Those are minor.

The major issue is that the Connection constructor is incorrect. It took me time to figure out what the correct way is to call the constructor.

So now to the solution of my version of the code. Unfortunately with async/await it is not as easy as simply adding "await". Intellisense complains as soon as I add await; it says that Main must be async. That is why I did the work in a separate function so that it can be async. However adding .Wait() seems to solve the problem except now I am getting _An error occurred while sending the request_. Is the following correct? If it is I don't know what the problem is.

    var vars = new Dictionary<string, object>
        {
            { "owner", "SimpleSamples" },
            { "name", "SimpleSamples.Repos" },
        };

Is that the correct syntax for name?

**Comment by SimpleSamples** [comment]
Okay the InnerException is:

    The request was aborted: Could not create SSL/TLS secure channel.

I found:

- [The request was aborted: Could not create SSL/TLS secure channel. · Issue #911 · googleapis/google-api-dotnet-client](https://github.com/googleapis/google-api-dotnet-client/issues/911)

- [The request was aborted: Could not create SSL/TLS secure channel. · Issue #1756 · octokit/octokit.net](https://github.com/octokit/octokit.net/issues/1756)

I do not understand all of those but I changed the target framework version to 4.6 and that error went away.

Then the result was null. So I tried using "Repos" for the name instead of SimpleSamples.Repos and that works. So that is another issue with the sample code that says "Rocks!".
